### PR TITLE
Create slaves from snapshots. configure num executors, ignore non-slave images

### PR DESCRIPTION
**Pull request by RoryHunter**
Fix an issue where you couldn't create slaves from an image snapshot, because droplet creation was using name slugs. Switch to image IDs.

Allow the number of executors per slave to be configured.

When checking how many slaves are running, filter droplets by their image name. This avoids counting droplets that are unrelated to Jenkins.

Sort lists of objects for getter UX when configuring slaves.

Some other internal refactoring.

### CODE REVIEW

File: src/main/java/com/dubture/jenkins/digitalocean/SlaveTemplate.java [hunk 1]
```diff
@@ -111,16 +112,18 @@
...
     @DataBoundConstructor
-    public SlaveTemplate(String imageId, String sizeId, String regionId, String idleTerminationInMinutes, String labelString) {
+    public SlaveTemplate(String imageId, String sizeId, String regionId, String idleTerminationInMinutes, String numExecutors, String labelString) {
         LOGGER.log(Level.INFO, "Creating SlaveTemplate with imageId = {0}, sizeId = {1}, regionId = {2}", new Object[] { imageId, sizeId, regionId});
         this.imageId = imageId;
         this.sizeId = sizeId;
         this.regionId = regionId;
 
         this.idleTerminationInMinutes = Integer.parseInt(idleTerminationInMinutes);
+        this.numExecutors = Integer.parseInt(numExecutors);
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/SlaveTemplate.java]: For all Templates created bevore applying this patch this value is null which results in a nullpointer exception. We might want to check for null and use 1 instead.

File: src/main/java/com/dubture/jenkins/digitalocean/Utils.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import static com.google.common.collect.Lists.newArrayList;
+
+/**
+ * Various utility methods that make it easier to obtain full lists of properties from Digital Ocean. Some API
+ * calls require page number, since the results are paginated, so these utilities will exhaust all the pages and
+ * return a single result set.
+ *
+ * @author Rory Hunter (rory.hunter@blackpepper.co.uk)
+ */
+public class Utils {
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/Utils.java]: Utils is kind of the wrong name for this class. It's rather DigitalOcean as static class

**Comment by anpieber** [comment]
First of all, thank you very much for the patch! There are only two minor issues, but especially the first can save some headages...

**Comment by RoryHunter** [comment]
Thanks for the feedback, I've pushed some fixes to our branch.

**Comment by anpieber** [comment]
Thank you very much for your changes. I'll merge and release them tomorrow!

**Comment by mfournier** [comment]
FYI I built the plugin from current master + this patchset, and everything seems to work fine (upgrading from latest release). Thanks a lot for your work @RoryHunter !

**Pull request merged**